packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts

```diff
@@ -99,22 +99,47 @@ type ValidateResponse = {
   errors: PipelineParserError[];
 };
 
+function getOrphanComments(comments: t.Comment[], visitedComments: Set<t.Comment>): t.Comment[] {
+  return comments.filter((comment) => !visitedComments.has(comment));
+}
+
+function getDisabledStagesFromEmptyPipeline(root: t.ArrayExpression) {
+  // Inner comments will only be available here if there is no other
+  // elements in the array, in our case it might mean that all stages
+  // are disabled
+  const isPipelineWithAllDisabledStages = root.elements.length === 0 && root.innerComments?.length;
+  if (!isPipelineWithAllDisabledStages) {
+    return [];
+  }
+
+  const innerComments = root.innerComments as t.Comment[];
+  const [visitedComments, stages] = extractStagesFromComments(innerComments);
+  // As all the stages are disabled and if we have comments that're
+  // at the end of the list, we need to make sure to attach them
+  // to the last stage as trailing comments.
+  const hasOrphanComments = stages.length > 0 && !visitedComments.has(innerComments[innerComments.length - 1]);
+  if (!hasOrphanComments) {
+    root.innerComments = innerComments.filter((node) => {
+      return !visitedComments.has(node);
+    });
+    return stages;
+  }
+
+  const lastStage = stages[stages.length - 1];
+  const orphanComments = getOrphanComments(innerComments, visitedComments);
+  lastStage.trailingComments = orphanComments;
+  adjustStageLoc(lastStage, (lastStage?.loc?.end.line ?? 0) + 1);
```

Comment:
Why do you need to adjust LOC here for the whole stage again?

Reply:
Not fixing the loc here, creates unexpected output with empty line. so used `adjustStageLoc` to have correct lines across.

Reply:
Should we do the realignment here for all the newly generated stages then maybe? I am a bit worried that if line alignment is needed here, doing only for one stage might be missing some corner case, also it's pretty cheap to do. Something like this maybe:

```ts
function adjustAllStagesLoc(stages: t.Expression[]) {
  stages.forEach((stage, idx) => {
    adjustStageLoc(stage, stages[idx - 1]?.loc?.end.line ?? 0);
  });
}

adjustAllStagesLoc(stages);
```

Reply:
Yes, no harm in adding that. But with other stages, as we do not attach anything manually, do we really need it? And if some edge case shows up, it may be unrelated.

In any case, i added this transformation as well (as its easy and cheap).

Reply:
All line order across stages is incorrect at this point, they all will have start line at 0 as every stage is parsed separately (comment stage parser resets source every time we find a valid stage), the `(lastStage?.loc?.end.line ?? 0) + 1` that was there before seems pretty arbitrary in general, but especially when aligning just one stage out of many, if we need this alignment to happen, a thing that would be easier to reason about in the future is to just align them all without exception and give then all a unified line order, especially as this is cheap to do. If this was not, then yes, it might've make sense to be a bit more careful about it